bin/node/src/service.rs

```diff
@@ -33,6 +35,24 @@ type FullClient = sc_service::TFullClient<Block, RuntimeApi, AlephExecutor>;
 type FullBackend = sc_service::TFullBackend<Block>;
 type FullSelectChain = sc_consensus::LongestChain<FullBackend, Block>;
 
+struct LimitNonfinalized(u32);
+
+impl<N: TryInto<u32> + Sub<Output = N>> BackoffAuthoringBlocksStrategy<N> for LimitNonfinalized {
+    fn should_backoff(
+        &self,
+        chain_head_number: N,
+        _chain_head_slot: Slot,
+        finalized_number: N,
+        _slow_now: Slot,
+        _logging_target: &str,
+    ) -> bool {
+        let nonfinalized_blocks = (chain_head_number - finalized_number)
+            .try_into()
+            .unwrap_or(u32::MAX);
```

Comment:
When reading this I wonder what if `chain_head_number - finalized_number` is negative and instead of `0` we output `u32::MAX` there? On the other hand, if `N` is a signed type then this can panic in some strange circumstances?

We should use `saturating_sub` instead of `-`.

Reply:
Fixed using Substrate's traits.